Library/Homebrew/formula_creator.rb

```diff
@@ -201,6 +201,18 @@ def install
             virtualenv_install_with_resources
         <% elsif @mode == :ruby %>
             ENV["GEM_HOME"] = libexec
+
+            # Install dependencies declared in a Gemfile.lock
+            # system "bundle", "config", "set", "without", "development", "test"
+            # system "bundle", "install"
+
+            # Install dependencies declared as resources
+            # resources.each do |r|
+            #   r.fetch
+            #   system "gem", "install", r.cached_download, "--ignore-dependencies",
+            #          "--no-document", "--install-dir", libexec
+            # end
+
             system "gem", "build", "\#{name}.gemspec"
             system "gem", "install", "\#{name}-\#{@version}.gem"
```

Comment:
This feels weird. Why is it needed?

Reply:
In order to make sure one of the previous commands (`bundle install` or resources) handles the dependencies.

Otherwise, for formulae with dependencies but without `Gemfile.lock`, this will install the latest gems and makes it less reproducible/checksummed. 

It is in our documentation:
https://github.com/Homebrew/brew/blob/638a3dc3a33c4c2029db058cf3d399008621e973/docs/Formula-Cookbook.md?plain=1#L248

Reply:
> Otherwise, for formulae with dependencies but without `Gemfile.lock`, this will install the latest gems and makes it less reproducible/checksummed.

This is true but I wonder how much of a problem this is for us in reality, particularly if we just move to using `bundle install` entirely instead (which I trust to handle the edge-cases here more than `gem install`).

I think the ideal outcome here (but not blocking this PR on that) would be to run `bundle install` and actually include the Gemfile.lock in e.g. the bottle/manifest/tab/prefix/etc.

We could take a similar approach for other packaging systems, too, and it avoids all the ~shit~ manual work required with resources.